plugins/source/stripe/client/testing.go

```diff
@@ -45,32 +43,33 @@ func MockTestHelper(t *testing.T, table *schema.Table, opts TestOptions) {
 		}
 	}()
 
-	newTestExecutionClient := func(ctx context.Context, logger zerolog.Logger, spec specs.Source, opts source.Options) (schema.ClientMeta, error) {
-		var stSpec Spec
-		if err := spec.UnmarshalSpec(&stSpec); err != nil {
-			return nil, fmt.Errorf("failed to unmarshal stripe spec: %w", err)
-		}
+	sched := scheduler.NewScheduler(scheduler.WithLogger(logger))
+	spec := &Spec{}
+	spec.SetDefaults()
+
+	cl := sclient.New("sk_test_myTestKey", getBackends(logger, *addr))
 
-		cl := sclient.New("sk_test_myTestKey", getBackends(logger, *addr))
-		return New(logger, spec, stSpec, cl, opts.Backend), nil
+	c := New(logger, *spec, cl, nil)
+
+	messages, err := sched.SyncAll(context.Background(), c, schema.Tables{table})
+	if err != nil {
+		t.Fatalf("failed to sync: %v", err)
+	}
+	records := filterInserts(messages).GetRecordsForTable(table)
+	emptyColumns := schema.FindEmptyColumns(table, records)
+	if len(emptyColumns) > 0 {
+		t.Fatalf("empty columns: %v", emptyColumns)
 	}
+}
 
-	p := source.NewPlugin(
-		table.Name,
-		version,
-		[]*schema.Table{
-			table,
-		},
-		newTestExecutionClient,
-	)
-	p.SetLogger(logger)
-	source.TestPluginSync(t, p, specs.Source{
-		Name:         "dev",
-		Path:         "cloudquery/dev",
-		Version:      version,
-		Tables:       []string{table.Name},
-		Destinations: []string{"mock-destination"},
-	})
+func filterInserts(msgs message.SyncMessages) message.SyncInserts {
+	inserts := []*message.SyncInsert{}
```

Comment:
I feel like this should be a part of SDK as I saw this in several PRs already

Reply:
it is, but it was renamed